exchanges/bitstamp/bitstamp_websocket.go

```diff
@@ -344,3 +411,47 @@ func (b *Bitstamp) seedOrderBook(ctx context.Context) error {
 	}
 	return nil
 }
+
+// fetchWSAuth Retrieves a userID and auth-token from REST for subscribing to a websocket channel
+// The token life-expectancy is only about 60s; use it immediately and do not store it
+func (b *Bitstamp) fetchWSAuth(ctx context.Context) (*websocketAuthResponse, error) {
```

Comment:
Assuming you're happy for it to be `sharedtestvalues.SkipTestIfCredentialsUnset` ?

Reply:
Yes please. A dedicated test and not just covered as part of another test. Also since its an API implementation, I think it should be exported

Reply:
- [x] Mock passing test
- [ ] Mock failing test ⛔ 
- [x] Live passing test manually
- [x] Changed to Public method

I've tried relatively hard do get a failing mock test but it would involve significant changes to mocks.
We request a fixed path, the only variables are the key and the signature, and the mocks ignore those values during matching.

A couple of options:
* Move the url to a var instead of const and change it to force the mock to use a different path
* Change the mock server

But for now I thought this is probably enough. It's more than we have for many other tests.

I've checked that we won't panic, etc with bad creds live.

88f65116a8b00daef112951ba15fc21b3f7cbc60

Reply:
A failing mock test is definitely not a requirement. I wouldn't even say a mock test is required for an authenticated function. Thank you for your efforts though, its still appreciated 🤙